tensorflow/java/src/test/java/org/tensorflow/op/PrimitiveOpTest.java

```diff
@@ -0,0 +1,177 @@
+import java.util.Set;
+
+import org.junit.Test;
+import org.tensorflow.Graph;
+import org.tensorflow.Operand;
+import org.tensorflow.Operation;
+import org.tensorflow.Output;
+import org.tensorflow.TestUtil;
+
+public class PrimitiveOpTest {
+
+  @Test
+  public void createOp() {
+    try (Graph g = new Graph()) {
+      Scope s = new Scope(g);
+      Output array = TestUtil.constant(g, "array", new int[2][2]);
+
+      TestOp test = TestOp.create(s, array);
+      assertNotNull(test.operation);
+
+      Output result = test.output();
+      assertNotNull(result);
+      assertSame(test.operation, result.op());
+      assertEquals(2, result.shape().size(0)); // 2 dims
+    }
+  }
+
+  @Test
+  public void createListOp() {
+    try (Graph g = new Graph()) {
+      Scope s = new Scope(g);
+      Output matrix = TestUtil.constant(g, "matrix", new int[2][2]);
+      Output array = TestUtil.constant(g, "array", new int[2]);
+
+      TestListOp test = TestListOp.create(s, Arrays.asList(matrix, array));
+      assertNotNull(test.operation);
+
+      List<Output> result = test.output();
+      assertNotNull(test.output());
+      assertEquals(2, result.size());
+      assertEquals(2, result.get(0).asOutput().shape().size(0)); // 2 dims
+      assertEquals(1, result.get(1).asOutput().shape().size(0)); // 1 dim
+    }
+  }
+
+  @Test
+  public void overrideOpName() {
+    try (Graph g = new Graph()) {
+      Scope s = new Scope(g);
+      Output array = TestUtil.constant(g, "array", new int[2]);
+
+      TestOp test1 = TestOp.create(s, array);
+      assertEquals("shape", test1.operation.name());
+
+      TestOp test2 = TestOp.create(s.withName("Test"), array);
+      assertEquals("Test", test2.operation.name());
+    }
+  }
+
+  @Test
+  public void equalsHashcode() {
+    try (Graph g = new Graph()) {
+      Scope s = new Scope(g);
+      Output array = TestUtil.constant(g, "array", new int[2]);
+
+      TestOp test1 = TestOp.create(s, array);
+      TestOp test2 = TestOp.create(s, array);
+      PrimitiveOp test3 = new PrimitiveOp(test1.operation) {};
+
+      // equals() tests
+      assertNotEquals(test1, test2);
+      assertEquals(test1, test3);
+      assertEquals(test3, test1);
+      assertNotEquals(test2, test3);
+
+      // hashcode() tests
+      Set<PrimitiveOp> ops = new HashSet<>();
+      assertTrue(ops.add(test1));
+      assertTrue(ops.add(test2));
+      assertFalse(ops.add(test3));
+    }
+  }
+
+  @Test
+  public void getOpString() {
+    try (Graph g = new Graph()) {
+      Scope s = new Scope(g);
+      Output array = TestUtil.constant(g, "array", new int[2]);
+
+      TestOp test = TestOp.create(s.withSubScope("tests"), array);
+      assertTrue(test.toString().contains("tests/shape"));
+      assertTrue(test.toString().contains("Shape"));
+    }
+  }
+
+  private static class TestOp extends PrimitiveOp implements Operand {
+
+    @Override
+    public Output asOutput() {
+      return output;
+    }
+
+    Output output() {
```

Comment:
Like I mentioned in a previous comment, It is to represent what will be generated automatically. Even if an operation implements `Operand`, it exposes its output with a named method. But that test class might go away anyway...